### PR TITLE
VMO-5604 Add class to match component name in root elements

### PR DESCRIPTION
**Pull request by laserpants**
Make root element class names match name of the component (kebab-cased) in accordance with coding style convention.

### CODE REVIEW

**Comment by safydy-r** [comment]
@laserpants , let's update the storyshots as many dom element has changed, and that affects the unit testing. You can check the [details](https://app.travis-ci.com/github/FLOIP/flow-builder/builds/249759265#L368) by clicking the Travis-ci details above.

Could you just run `yarn test:unit --no-cache -u` to update storyshots, and commit push after?

**Note**: For now, we only have `InteractionDesigner/Toolbar` storyshot, the other storyshots (blocks, flow) don't show for some reason (don't worry about these missing storyshots here)

**Comment by laserpants** [comment]
@safydy-r Done! :checkered_flag: Please see [0a8880b63ccde8ff157cd75ebb0c5aaddb5a6a08](https://github.com/FLOIP/flow-builder/pull/229/commits/0a8880b63ccde8ff157cd75ebb0c5aaddb5a6a08).

File: src/components/interaction-designer/block-editors/Categorization.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="block-categorization">
+  <div class="categorization block-categorization">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/Categorization.vue:2]: `block-categorization` class can be removed.
It's probably from the old times when the names of the components in the /block-editor folder started with `block-...`.
And there is no associated css in the flow-builder or voto5 repo.

File: src/components/interaction-designer/block-editors/MaximumChoicesEditor.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/maximum_choices`">
-    <div class="block-validation-max-choices">
+    <div class="maximum-choices-editor block-validation-max-choices">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/MaximumChoicesEditor.vue:5]: the same for `block-validation-max-choices`

File: src/components/interaction-designer/block-editors/MaxDigitEditor.vue [hunk 3]
```diff
@@ -3,7 +3,7 @@
     v-if="hasIvr"
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/ivr/max_digits`">
-    <div class="block-max-digits">
+    <div class="max-digit-editor block-max-digits">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/MaxDigitEditor.vue:6]: and `block-max-digits`

File: src/components/interaction-designer/block-editors/MaximumNumericEditor.vue [hunk 4]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/validation_maximum`">
-    <div class="block-validation-max">
+    <div class="maximum-numeric-editor block-validation-max">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/MaximumNumericEditor.vue:5]: and `block-validation-max`

File: src/components/interaction-designer/block-editors/MinimumChoicesEditor.vue [hunk 5]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/minimum_choices`">
-    <div class="block-validation-min-choices">
+    <div class="minimum-choices-editor block-validation-min-choices">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/MinimumChoicesEditor.vue:5]: and `block-validation-min-choices`

File: src/components/interaction-designer/block-editors/MinimumNumericEditor.vue [hunk 6]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/validation_minimum`">
-    <div class="block-validation-min">
+    <div class="minimum-numeric-editor block-validation-min">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/MinimumNumericEditor.vue:5]: and `block-validation-min`

File: src/components/interaction-designer/block-editors/SemanticLabelEditor.vue [hunk 7]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/semantic_label`">
-    <div class="block-semantic-label">
+    <div class="semantic-label-editor block-semantic-label">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/SemanticLabelEditor.vue:5]: and `block-semantic-label`

File: src/components/interaction-designer/block-editors/TagSelector.vue [hunk 8]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{isValid}"
     :message-key="`block/${block.uuid}/tags`">
-    <div class="block-group">
+    <div class="tag-selector block-group">
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/TagSelector.vue:5]: and `block-group` (another one)

File: src/components/interaction-designer/clipboard/shared/BlockActionButtons.vue [hunk 9]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div v-if="isFocused" class="row blockActionButtonRoot">
+  <div v-if="isFocused" class="block-action-buttons row blockActionButtonRoot">
```
Comment (bulatgab) [inline src/components/interaction-designer/clipboard/shared/BlockActionButtons.vue:2]: Maybe we could reuse the new `block-action-buttons` class that you added for styles as well?
Especially, as blockActionButtonRoot is not kebab-case.

File: src/components/interaction-designer/clipboard/ClipboardRoot.vue [hunk 10]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="clipboard-root-view">
+  <div class="clipboard-root clipboard-root-view">
```
Comment (bulatgab) [inline src/components/interaction-designer/clipboard/ClipboardRoot.vue:2]: Maybe we could reuse the newly added `clipboard-root` for styles (and remove the `clipboard-root-view`)?

**Comment by bulatgab** [comment]
Probably, `flow-editor` class shouldn't be in `ExitBlockEditorToggle.vue`. It was added by mistake (copy-pasted?), most likely. Again, no css anywhere.

File: src/components/interaction-designer/flow-editors/FlowLabelEditor.vue [hunk 11]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="form-group flow-label">
+  <div class="flow-label-editor form-group flow-label">
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/FlowLabelEditor.vue:2]: and `flow-label`

File: src/components/interaction-designer/flow-editors/FirstBlockEditorButton.vue [hunk 12]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="starting-block-button">
+  <div class="first-block-editor-button starting-block-button">
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/FirstBlockEditorButton.vue:2]: and `starting-block-button`

File: src/components/interaction-designer/flow-editors/InteractionTimeoutEditor.vue [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="form-group flow-interaction-timeout">
+  <div class="interaction-timeout-editor form-group flow-interaction-timeout">
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/InteractionTimeoutEditor.vue:2]: and `flow-interaction-timeout`

File: src/components/interaction-designer/flow-editors/ModesEditor.vue [hunk 14]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="form-group flow-modes">
+  <div class="modes-editor form-group flow-modes">
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/ModesEditor.vue:2]: and `flow-modes`

File: src/components/interaction-designer/toolbar/ErrorNotifications.vue [hunk 15]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main class="error-notifications-wrapper">
+  <main class="error-notifications error-notifications-wrapper">
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/ErrorNotifications.vue:2]: we probably don't need `error-notifications-wrapper` either

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue [hunk 16]
```diff
@@ -2,7 +2,7 @@
   <div
     ref="builder-toolbar"
     class="tree-builder-toolbar d-flex flex-column">
-    <div class="tree-builder-toolbar-main-menu d-flex flex-column">
+    <div class="tree-builder-toolbar tree-builder-toolbar-main-menu d-flex flex-column">
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:5]: This can be undone. There is already a root element with the class we need.

**Review by bulatgab — CHANGES_REQUESTED** [review]
Clean up classes and styles. See comments [here](https://github.com/FLOIP/flow-builder/pull/229/files) and in a separate [comment](https://github.com/FLOIP/flow-builder/pull/229#issuecomment-1108932410)

File: src/components/interaction-designer/block-editors/Categorization.vue (hunk 1 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/Categorization.vue:2]: :heavy_check_mark: Done!

File: src/components/interaction-designer/block-editors/MaximumChoicesEditor.vue (hunk 2 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/MaximumChoicesEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/FormatStringEditor.vue [hunk 17]
```diff
@@ -2,7 +2,7 @@
   <validation-message
     #input-control="{ isValid }"
     :message-key="`block/${block.uuid}/config/format_string`">
-    <div class="block-format-string">
+    <div class="format-string-editor block-format-string">
```
[the remark this answers is not in the document]
Comment (laserpants) [reply src/components/interaction-designer/block-editors/FormatStringEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/MaximumNumericEditor.vue (hunk 4 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/MaximumNumericEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/MinimumChoicesEditor.vue (hunk 5 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/MinimumChoicesEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/MinimumNumericEditor.vue (hunk 6 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/MinimumNumericEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/SemanticLabelEditor.vue (hunk 7 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/SemanticLabelEditor.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/block-editors/TagSelector.vue (hunk 8 above)
Comment (laserpants) [reply src/components/interaction-designer/block-editors/TagSelector.vue:5]: :heavy_check_mark:

File: src/components/interaction-designer/toolbar/ErrorNotifications.vue (hunk 15 above)
Comment (laserpants) [reply src/components/interaction-designer/toolbar/ErrorNotifications.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/clipboard/shared/BlockActionButtons.vue (hunk 9 above)
Comment (laserpants) [reply src/components/interaction-designer/clipboard/shared/BlockActionButtons.vue]: :heavy_check_mark:

File: src/components/interaction-designer/clipboard/ClipboardRoot.vue (hunk 10 above)
Comment (laserpants) [reply src/components/interaction-designer/clipboard/ClipboardRoot.vue]: :heavy_check_mark:

File: src/components/interaction-designer/flow-editors/FlowLabelEditor.vue (hunk 11 above)
Comment (laserpants) [reply src/components/interaction-designer/flow-editors/FlowLabelEditor.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/flow-editors/FirstBlockEditorButton.vue (hunk 12 above)
Comment (laserpants) [reply src/components/interaction-designer/flow-editors/FirstBlockEditorButton.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/flow-editors/InteractionTimeoutEditor.vue (hunk 13 above)
Comment (laserpants) [reply src/components/interaction-designer/flow-editors/InteractionTimeoutEditor.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/flow-editors/LanguagesEditor.vue [hunk 18]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="form-group flow-languages">
+  <div class="languages-editor form-group flow-languages">
```
[the remark this answers is not in the document]
Comment (laserpants) [reply src/components/interaction-designer/flow-editors/LanguagesEditor.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/flow-editors/ModesEditor.vue (hunk 14 above)
Comment (laserpants) [reply src/components/interaction-designer/flow-editors/ModesEditor.vue:2]: :heavy_check_mark:

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue (hunk 16 above)
Comment (laserpants) [reply src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue]: :heavy_check_mark:

**Comment by laserpants** [comment]
@bulatgab I have updated the PR now based on your feedback.

**Review by safydy-r — APPROVED** [review]
thanks @laserpants for this, I'm approving !
I do think we still need to QA this because we edited the npm dependencies. My suggestion for that is:
- you create an new branch in `FLOIP:flow-builder`, because QA won't be able to test the fork repo
- edit this PR's destination branch to that new branch
- once QA-ed, you create a new PR targeting master, you can merge it without asking approval by putting this original PR's url as approval reference. 

What do you think ?

**Comment by laserpants** [comment]
This PR is superseded by https://github.com/FLOIP/flow-builder/pull/233.

**Pull request closed without merging**